### ISSUE TITLE
tc(class/filter) leak when use kubenet

### ISSUE THREAD

**Issue by hongli-my** [kubernetes/kubernetes#56840]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kink bug

**What happened**:
======pod.yaml=======
```
apiVersion: v1
kind: Pod
metadata:
  name: qos
  annotations:
    kubernetes.io/ingress-bandwidth: 2M
    kubernetes.io/egress-bandwidth: 3M
spec:
  containers:
  - name: test
    image: moutten/iperf
```
create pod will create tc class and filter
```
tc class add dev cbr0 parent 1: classid 1:1 htb rate 2000kbit
tc filter add dev cbr0 protocol ip parent 1:0 prio 1 u32 match ip dst 10.6.0.71/32 flowid 1:1
tc class add dev cbr0 parent 1: classid 1:2 htb rate 3000kbit
tc filter add dev cbr0 protocol ip parent 1:0 prio 1 u32 match ip src 10.6.0.71/32 flowid 1:2
```
delete pod only delete one class and filter
```
tc filter del dev cbr0 parent 1: proto ip prio 1 handle 800::800 u32
tc class del dev cbr0 parent 1: classid 1:1
```
use `tc class show dev cbr0` or `tc filter show dev cbr0` also have some rule does not delete

Maybe a bug or some where need to be config in kernel ?

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
the func findCIDRClass only find one, maybe need to return slice ?

```
func (t *tcShaper) findCIDRClass(cidr string) (class, handle string, found bool, err error) {
	data, err := t.e.Command("tc", "filter", "show", "dev", t.iface).CombinedOutput()
	if err != nil {
		return "", "", false, err
	}

	hex, err := hexCIDR(cidr)
	if err != nil {
		return "", "", false, err
	}
	spec := fmt.Sprintf("match %s", hex)

	scanner := bufio.NewScanner(bytes.NewBuffer(data))
	filter := ""
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		if len(line) == 0 {
			continue
		}
		if strings.HasPrefix(line, "filter") {
			filter = line
			continue
		}
		if strings.Contains(line, spec) {
			parts := strings.Split(filter, " ")
			// expected tc line:
			// filter parent 1: protocol ip pref 1 u32 fh 800::800 order 2048 key ht 800 bkt 0 flowid 1:1
			if len(parts) != 19 {
				return "", "", false, fmt.Errorf("unexpected output from tc: %s %d (%v)", filter, len(parts), parts)
			}
			return parts[18], parts[9], true, nil
		}
	}
	return "", "", false, nil
}
```
**Environment**:
- Kubernetes version (use `v1.7.9`):
- Cloud provider or hardware configuration:
- OS (Ubuntu 16.04 LTS):
- Kernel (e.g. `4.4.0-31-generic`):
- Install tools:
- Others:

**Comment by hongli-my** [comment]
sig/network

**Comment by cmluciano** [comment]
@kubernetes/sig-network-bugs

**Comment by m1093782566** [comment]
> the func findCIDRClass only find one, maybe need to return slice ?

xref: `GetCIDRs()` in the same file with func findCIDRClass.

**Comment by m1093782566** [comment]
/close 

change is available in #58687.
